### DISCUSSION TITLE
DAI 2023-2024 - Practical work 1 - Sarah Jallon

### DISCUSSION THREAD

**Discussion by sarsurgithub** [heig-vd-dai-course/heig-vd-dai-course#82] [Show and tell]
For this practical work I will create an application that takes a text as an argument and will produce an ascii art as a result (written in a file). This ascii art will be the text in a big font using ascii characters.
If this is easy and I have enough time I'd love to change my program so that it takes a picture as an input and gives an ascii art of the picture as an output.
For this I found the following ressources: [res1](https://stackoverflow.com/questions/56100461/how-to-draw-an-image-using-ascii-symbols) and [res2](https://www.baeldung.com/ascii-art-in-java)
I still have to look into it but this looks promising!

**Comment by ludelafo** [comment]
Great! Looks promising!

I see that some people mentioned the usage of UTF-8 characters in the resources you posted. It could be some optional arguments to the CLI to set the output charset as well.

**Comment by sarsurgithub** [comment]
@ludelafo @hadrylouis Hi! I have a question regarding the gitignore for the practical work, I thought it would be better to put *.txt in my gitignore in order not to have the generated files from my program appear on github.
But this criterion makes me doubt it: The repository contains test files to use with the CLI
This means I should provide some input files already completed ? And in that case how can make sure only the input files I prepared stay on my repo and not the output files / other not relevant input files do stay on my repo ? 
Thanks!

**Reply by ludelafo** [reply]
Hello!

We think it is a good practice to add binary and/or output files to your gitignore file as they generally take space and can be generated automatically.

In your case, you could add a directory `examples` containing all the examples you would like to display (input `.txt` files for example). In your gitignore file, you would have something similar to this (disclaimer: I haven't tested the following gitignore, but it should work):

```gitignore
# Ignore all text files inside the repository
*.txt

# Except the input files in the `example` directory
!examples/**/*_input.txt
```

This would allow all other text files but the input ones you know you really want to keep in the repository ignored by Git.

In my opinion, the examples could contain the outputs as well to allow someone interested in your work to check the results without having to run your CLI locally. You can decide what you prefer.

You can find more information in the official documentation here: <https://git-scm.com/docs/gitignore#_examples>.

Does it answer your question?

**Reply by sarsurgithub** [reply]
Thanks! It absolutely answers my question, I was not aware of the !syntax and it is really useful!

**Comment by sarsurgithub** [comment]
Here is my [repo](https://github.com/sarsurgithub/DAI-Practical-Work-1)
And the last commit hash is 8209c34d0a5617b40899b0ab5469c512f341551a
@ludelafo @hadrylouis
